lib/buffer.js

```diff
@@ -637,6 +637,28 @@ Buffer.prototype.includes = function includes(val, byteOffset, encoding) {
   return this.indexOf(val, byteOffset, encoding) !== -1;
 };
 
+// Limit currently not implemented; reserved
+Buffer.prototype.split = function split(sep, limit) {
+  if (!limit)
+    limit = null;
+  if (!sep)
+    return [this];
```

Comment:
This should simply be `if (sep === undefined)`. Allow everything else to be passed to C++, checked and thrown if necessary. We don't want strange values like `NaN` not causing this to throw but `true` to throw.